File: 2022/day_1/src/main.rs
#![allow(dead_code, unused_variables, unused_mut)]

use aoc_std::*;

fn part1(input_file: String) -> usize {
    let mut sum = 0;
    let mut highest = 0;
    for line in input_file.split('\n') {
        if !line.is_empty() {
            sum += line.parse::<usize>().unwrap_or(0);
        } else {
            if sum > highest {
                highest = sum;
            }
            sum = 0;
        }
    }
    highest
}

fn part2(input_file: String) -> usize {
    let mut sum = 0;
    let mut highest_list = vec![];
    for line in input_file.split('\n') {
        if !line.is_empty() {
            sum += line.parse::<usize>().unwrap_or(0);
        } else {
            highest_list.push(sum);
            sum = 0;
        }
    }
    highest_list.sort();
    highest_list.reverse();
    highest_list[0] + highest_list[1] + highest_list[2]
}

fn main() {
    let part = 2;

    let input_file = load_file("input.txt");

    if part == 1 {
        print_answer(part1(input_file));
    } else if part == 2 {
        print_answer(part2(input_file));
    }
}

File: 2022/day_1/main.lua
local function lines_from(file)
    local lines = {}
        for line in io.lines(file) do
            lines[#lines + 1] = line
        end
    return lines
end

local function part1(lines)
    local sum, highest = 0, 0
    for _, line in pairs(lines) do
        if line ~= "" then
            sum = sum + line
        else
            if sum > highest then
                highest = sum
            end
            sum = 0
        end
    end
    print(highest)
end

local function part2(lines)
    local sum, highest = 0, {}
    for _, line in pairs(lines) do
        if line ~= "" then
            sum = sum + line
        else
            table.insert(highest, sum)
            sum = 0
        end
    end
    table.insert(highest, sum)
    table.sort(highest)
    local final = highest[#highest] + highest[#highest-1] + highest[#highest-2]
    print(final)
end

local function main()
    local part = 2

    local file_lines = lines_from("input.txt")
    if part == 1 then
        part1(file_lines)
    else
        part2(file_lines)
    end

end

main()
